crates/win-api-wrappers/src/service.rs

```diff
@@ -0,0 +1,150 @@
+
+        // SAFETY: Handle returned by `OpenServiceW` is valid and needs to be closed after use,
+        // thus it is safe to take ownership of it via `Owned`.
+        let handle = unsafe { Owned::new(raw_service_handle) };
+
+        Ok(Service { handle })
+    }
+
+    pub fn open_service_read(&self, service_name: &str) -> anyhow::Result<Service> {
+        self.open_service_with_access(service_name, SERVICE_QUERY_CONFIG | SERVICE_QUERY_STATUS)
+    }
+
+    pub fn open_service_all_access(&self, service_name: &str) -> anyhow::Result<Service> {
+        self.open_service_with_access(service_name, SERVICE_ALL_ACCESS)
+    }
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum ServiceStartupMode {
+    Boot,
+    System,
+    Automatic,
+    Manual,
+    Disabled,
+}
+
+pub struct Service {
+    handle: Owned<SC_HANDLE>,
+}
+
+impl Service {
+    pub fn startup_mode(&self) -> anyhow::Result<ServiceStartupMode> {
+        let mut cbbufsize = 0u32;
+        let mut pcbbytesneeded = 0u32;
+
+        // SAFETY: No preconditions. Query required buffer size.
+        let result = unsafe { QueryServiceConfigW(*self.handle, None, 0, &mut pcbbytesneeded) };
+
+        match result {
+            Err(err) if err.code() == ERROR_INSUFFICIENT_BUFFER.to_hresult() => {
+                // Expected error, continue.
+            }
+            Err(err) => return Err(err.into()),
+            Ok(_) => panic!("QueryServiceConfigW should fail with ERROR_INSUFFICIENT_BUFFER"),
+        }
+
+        // The most typical buffer we work with in Rust are homogeneous arrays of integers such
+        // as [u8] or Vec<u8>, but in Microsoft’s Win32 documentation, a `buffer` generally refers
+        // to a caller-allocated memory region that an API uses to either input or output data, and
+        // it is ultimately coerced into some other type with various alignment requirements.
+        //
+        // lpServiceConfig should point to aligned buffer that could hold a QUERY_SERVICE_CONFIGW
+        // structure.
+        let layout = Layout::from_size_align(
+            usize::try_from(pcbbytesneeded).expect("pcbbytesneeded < 8K as per MSDN"),
+            align_of::<QUERY_SERVICE_CONFIGW>(),
+        )?;
+
+        // SAFETY: The layout initialization is checked using the Layout::from_size_align method.
+        let mut buffer = unsafe { RawBuffer::alloc_zeroed(layout)? };
+
+        // SAFETY: Buffer passed to `lpServiceConfig` have enough size to hold a
+        // QUERY_SERVICE_CONFIGW structure, as required size was queried and allocated above.
+        // Passed buffer have correct alignment to hold QUERY_SERVICE_CONFIGW structure.
+        unsafe {
+            // NOTE: Pointer cast is valid, as `buffer` is allocated with correct alignment above.
+            #[allow(clippy::cast_ptr_alignment)]
+            QueryServiceConfigW(
+                *self.handle,
+                Some(buffer.as_mut_ptr().cast::<QUERY_SERVICE_CONFIGW>()),
```

Comment:
suggestion: Put the comment to justify the suppression on the same line. This avoid confusion with other unrelated comments.

suggestion: Use `expect` instead of `allow`, in case the lint is not fired anymore some day

nitpick: In fact, the `NOTE:` prefix is rarely meaningful.

```suggestion
            #[expect(clippy::cast_ptr_alignment)] // Pointer cast is valid, as `buffer` is allocated with correct alignment above.
            QueryServiceConfigW(
                *self.handle,
                Some(buffer.as_mut_ptr().cast::<QUERY_SERVICE_CONFIGW>()),
```